plugins/lookup_harmony/__init__.py

```diff
@@ -0,0 +1,135 @@
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.
+
+PLUGIN_NAME = "Look Up with Harmony"
+PLUGIN_AUTHOR = "Jade"
+PLUGIN_DESCRIPTION = """
+Adds a right click option to look up releases on Harmony using the GTIN code.
+If a MusicBrainz release ID is available, it will be included in the lookup URL.
+"""
+PLUGIN_VERSION = "0.1.0"
+PLUGIN_API_VERSIONS = ["2.10"]  # Only tested in 2.13 but should be fine
+PLUGIN_LICENSE = "GPL-2.0"
+PLUGIN_LICENSE_URL = "https://www.gnu.org/licenses/gpl-2.0.txt"
+
+from picard import log
+from picard.album import Album
+from picard.cluster import Cluster
+from picard.track import Track
+from picard.ui.itemviews import (
+    BaseAction,
+    register_album_action,
+    register_cluster_action,
+    register_track_action,
+)
+from picard.util.webbrowser2 import open as web_open
+
+
+def perform_harmony_lookup(metadata):
+    """Perform a Harmony lookup using the metadata from an album, track, or cluster."""
+    try:
+        barcode_value = metadata.get("barcode", "")
+        log.debug(
+            "{0}: barcode tag raw content: {1} (type: {2})".format(
+                PLUGIN_NAME, repr(barcode_value), type(barcode_value).__name__
+            )
+        )
+
+        # Convert to string and strip (handle lists, etc.)
+        if isinstance(barcode_value, list):
+            gtin = str(barcode_value[0]).strip() if barcode_value else ""
+        else:
+            gtin = str(barcode_value).strip() if barcode_value else ""
+
+        # Fallback to 'upc' tag if barcode is not available
+        if not gtin:
+            upc_value = metadata.get("upc", "")
+            log.debug(
+                "{0}: upc tag raw content: {1} (type: {2})".format(
+                    PLUGIN_NAME, repr(upc_value), type(upc_value).__name__
+                )
+            )
+
+            if isinstance(upc_value, list):
+                gtin = str(upc_value[0]).strip() if upc_value else ""
+            else:
+                gtin = str(upc_value).strip() if upc_value else ""
+
+        if not gtin:
+            log.warning(
+                "{0}: No GTIN/UPC code found for this release.".format(PLUGIN_NAME)
+            )
+            return
+
+        log.info("{0}: Using GTIN: {1}".format(PLUGIN_NAME, gtin))
+
+        mbid_value = metadata.get("musicbrainz_albumid", "")
+        log.debug(
+            "{0}: musicbrainz_albumid tag raw content: {1} (type: {2})".format(
+                PLUGIN_NAME, repr(mbid_value), type(mbid_value).__name__
+            )
+        )
+
+        if isinstance(mbid_value, list):
+            musicbrainz_albumid = str(mbid_value[0]).strip() if mbid_value else ""
+        else:
+            musicbrainz_albumid = str(mbid_value).strip() if mbid_value else ""
+
+        # Build the Harmony URL
+        url = "https://harmony.pulsewidth.org.uk/release?gtin={0}&deezer=&itunes=&spotify=&tidal=&musicbrainz={1}".format(
```

Comment:
Two quick thoughts here:
1. Providers shouldn't be hard-coded, there are provider categories (currently allowed values: default, all, preferred) which can be used instead:
```suggestion
        url = "https://harmony.pulsewidth.org.uk/release?gtin={0}&category=preferred&musicbrainz={1}".format(
```
2. It might make sense to make the base URL configurable as well. Although most people use the pulsewidth server, alternative instances could then be used.